proxy/logging/LogField.cc

```diff
@@ -772,7 +772,9 @@ LogFieldList::marshal_len(LogAccess *lad)
   int bytes = 0;
   for (LogField *f = first(); f; f = next(f)) {
     if (f->type() != LogField::sINT) {
-      bytes += f->marshal_len(lad);
+      const int len = f->marshal_len(lad);
```

Comment:
Wouldn't adding a check on {{len > 0}} here and not add it to bytes also work?  This way you wouldn't have to set m_cache_lookup_url_canon_str or have INVALID_STR.